spring-webflux/src/test/java/org/springframework/web/reactive/function/MultipartIntegrationTests.java

```diff
@@ -95,12 +103,29 @@ void parts(HttpServer httpServer) throws Exception {
 				.verify(Duration.ofSeconds(5));
 	}
 
-	@ParameterizedHttpServerTest
+	static Stream<HttpServer> httpServersWithoutUndertow() {
+		return Stream.of(
+				new JettyHttpServer(),
+				new ReactorHttpServer(),
+				new TomcatHttpServer()
+		);
+	}
+
+	@ParameterizedTest(name = "[{index}] {0}")
+	@MethodSource("httpServersWithoutUndertow")
 	void transferTo(HttpServer httpServer) throws Exception {
 		// TODO: check why Undertow fails
-		if (httpServer instanceof UndertowHttpServer) {
-			return;
-		}
```

Comment:
Instead of recreating all of the parameterized test infrastructure, please change the above part that you removed to an assumption that fails when the server is Undertow -- for example, with JUnit Jupiter's `assumeFalse(...)` (if Checkstyle permits that) or AssertJ's `Assumptions` class. For an example, see `org.springframework.http.server.reactive.ZeroCopyIntegrationTests.zeroCopy(HttpServer)`.

Once you're done that, please revert all of the unnecessary changes in this test class.

Reply:
Using `assumeTrue` in a catch-block is not what I had in mind, since that will hide all failures (for all server types), which is unacceptable.

In https://github.com/spring-projects/spring-framework/commit/a247b83cd9c9aefd3c329d493c5ce7cd11d0cdfa you can see what I meant: effectively we want to _abort_ the test only if the current server is Undertow.

Thus, please rebase your work on `main` and limit the changes to the new `transferToWithUndertow()` test method and the extracted `verifyTransferTo(HttpServer)` method (without the try-catch).

`transferToWithUndertow()` will need to be annotated with `@Disabled` for the time being.

In addition, please add `See gh-25310` to your commit message so that it links to #25310.